README.md

```diff
@@ -1,2 +1,133 @@
-# sdm-versions-manager
-Creation of a SDM database of data models’ versions with information of the current version of the data model and the corresponding hash code in GitHub repository.
+# SDM Versions Manager
+
+## Overview
+
+**sdm-versions-manager** is a Python-based service designed to create a database of Smart Data Models (SDMs) versions.
+
+ It retrieves information about the current version of each data model and its corresponding hash code from a GitHub repository. 
+ 
+ This service aims to facilitate the management and tracking of data model versions, ensuring that users have access to the most up-to-date information.
+
+## Features
+
+- Fetches historical commit data from GitHub for specified data models.
+- Extracts version information and commit hashes from the schema files.
+- Populates a MongoDB database with the current versions of data models.
+- Prevents duplicate entries in the database by checking existing records.
+
+## Requirements
+
+- Python 3.11 or higher
+- MongoDB (local or hosted)
+- GitHub personal access token (for API access)
+- Required Python packages (listed in `requirements.txt`)
+
+## Installation 
+
+1. **Clone the repository**:
+
+   ```bash
+   git clone https://github.com/yourusername/sdm-versions-manager.git
+   cd sdm-versions-manager
+   ```
+
+2. **Create a virtual environment (optional but recommended)**:
+    ```bash
+    python -m venv venv
+    source venv/bin/activate
+    ```  
+
+3. **Install the required packages**:
+    ```bash
+    pip install -r requirements.txt
+    ```
+
+4. **Set up environment variables**:
+
+    Create a `.env` file in the root directory of your project with the following content:
+
+    ```text
+    GITHUB_TOKEN=your_github_token
+    MONGO_URI=mongodb://localhost:27017/
+    DB_NAME=your_database_name
+    COLLECTION_NAME=your_collection_name
+    ```
+
+4. **Configure the data models**:
+
+    Create a `config.json` file in the root directory with the following structure that containes all the data models used in WATERVERSE:
```

Comment:
Could you elaborate more on what you mean here? 
1. Should I justify why we have this structure of (subject, data model) 
2. Should I say that these data models were derived from digital twin data from Pilots?

Reply:
Just why are we selecting those data models and in case that it is due to the project, then you should explain that for example we are selecting those initial data models.

Keep in mind that this is a general service, but at the moment we are using for WATERVERSE data models but it is expected that in the future will be used for others or all data models.